tests/e2e/monitoring/test_mon_log_trimming.py

```diff
@@ -0,0 +1,171 @@
+            self.ct_pod.exec_ceph_cmd(
+                ceph_cmd=f"ceph tell mon.{self.selected_mon_pod} injectargs "
+                f"--paxos_service_trim_min={self.PAXOS_MIN_DEFAULT}"
+            )
+            self.ct_pod.exec_ceph_cmd(
+                ceph_cmd=f"ceph tell mon.{self.selected_mon_pod} injectargs "
+                f"--paxos_service_trim_max={self.PAXOS_MAX_DEFAULT}"
+            )
+            self.ct_pod.exec_ceph_cmd(
+                ceph_cmd=f"ceph tell mon.{self.selected_mon_pod} injectargs "
+                f"--osd_op_complaint_time={self.PAXOS_COMPLAINT_TIME_DEFAULT}"
+            )
+
+        request.addfinalizer(finalizer)
+
+    def check_mon_db_trim(self):
+        """
+        Check mon db size while fio runs in the background
+        """
+        while not self.stop_checking_mon_db:
+            temp_mon_db_size = self.get_mon_db_size_in_kb()
+            assert temp_mon_db_size is not None, "Failed to get mon db size"
+            log.info(
+                f"Monitoring mon-{self.selected_mon_pod} db size: {temp_mon_db_size}K"
+            )
+
+            if temp_mon_db_size < self.current_mon_db_size:
+                log.info(
+                    f"Mon db trimmed from {self.current_mon_db_size} to {temp_mon_db_size}"
+                )
+                self.mon_db_trim_count = self.mon_db_trim_count + 1
+            self.current_mon_db_size = temp_mon_db_size
+        log.info(f"Number of trims performed: {self.mon_db_trim_count}")
+
+    def get_mon_db_size_in_kb(self):
+        """
+        Get mon db size and returns the size in KB
+        The output of 'du -sh' command contains the size of the directory and its path as string
+        e.g. "67M\t/var/lib/ceph/mon/ceph-c/store.db"
+        The size is extracted by splitting the string with '\t'.
+        The size format for example: 1K, 234M, 2G
+        For uniformity, this test uses KB
+        """
+        size = self.selected_mon_pod_obj.exec_cmd_on_pod(
+            f"du -sh /var/lib/ceph/mon/ceph-{self.selected_mon_pod}/store.db",
+            out_yaml_format=False,
+        )
+        size = re.split("\t+", size)
+        assert len(size) > 0, f"Failed to get mon-{self.selected_mon_pod} db size"
+        size = size[0]
+
+        if "K" in size:
+            return size[:-1]
+        elif "M" in size:
+            size = size[:-1]
+            return int(size) * 1024
+        elif "G" in size:
+            size = size[:-1]
+            return int(size) * 1024 * 1024
+        return None
+
+    def test_mon_log_trimming(self):
+        """
+        Check that mon db actually get trimmed
+
+        """
+        self.initial_db_size = self.get_mon_db_size_in_kb()
+        log.info(f"Initial db size: {self.initial_db_size}K")
+        self.fio_pod_obj.run_io(
+            storage_type="fs",
+            size="1GB",
+            runtime=300,
+        )
+        thread1 = threading.Thread(target=self.check_mon_db_trim)
+        thread1.start()
+
```

Comment:
Is the step to delete the osd pod missing?

Reply:
Done